test/java/org/chromium/net/testing/Http2TestServerTest.java

```diff
@@ -0,0 +1,202 @@
+package org.chromium.net.testing;
+
+import static io.envoyproxy.envoymobile.engine.EnvoyConfiguration.TrustChainVerification.ACCEPT_UNTRUSTED;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.chromium.net.testing.CronetTestRule.SERVER_CERT_PEM;
+import static org.chromium.net.testing.CronetTestRule.SERVER_KEY_PKCS8_PEM;
+
+import android.content.Context;
+import androidx.test.core.app.ApplicationProvider;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import io.envoyproxy.envoymobile.AndroidEngineBuilder;
+import io.envoyproxy.envoymobile.Engine;
+import io.envoyproxy.envoymobile.EnvoyError;
+import io.envoyproxy.envoymobile.RequestHeaders;
+import io.envoyproxy.envoymobile.RequestHeadersBuilder;
+import io.envoyproxy.envoymobile.RequestMethod;
+import io.envoyproxy.envoymobile.ResponseHeaders;
+import io.envoyproxy.envoymobile.ResponseTrailers;
+import io.envoyproxy.envoymobile.UpstreamHttpProtocol;
+import io.envoyproxy.envoymobile.engine.AndroidJniLibrary;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(AndroidJUnit4.class)
+public class Http2TestServerTest {
+
+  private Engine engine;
+
+  @BeforeClass
+  public static void loadJniLibrary() {
+    AndroidJniLibrary.loadTestLibrary();
+  }
+
+  @Before
+  public void setUpEngine() throws Exception {
+    CountDownLatch latch = new CountDownLatch(1);
+    Context appContext = ApplicationProvider.getApplicationContext();
+    engine = new AndroidEngineBuilder(appContext)
+                 .setTrustChainVerification(ACCEPT_UNTRUSTED)
+                 .setOnEngineRunning(() -> {
+                   latch.countDown();
+                   return null;
+                 })
+                 .build();
+    Http2TestServer.startHttp2TestServer(appContext, SERVER_CERT_PEM, SERVER_KEY_PKCS8_PEM);
+    latch.await(); // Don't launch a request before initialization has completed.
+  }
+
+  @After
+  public void shutdown() throws Exception {
+    engine.terminate();
+    Http2TestServer.shutdownHttp2TestServer();
+  }
+
+  @Test
+  public void get_schemeIsHttps() throws Exception {
```

Comment:
Should this be named getSchemeIsHttps()? (or is this some framework which requires this unusual name)

Reply:
Agreed, this name strikes me as unusual.

Reply:
Done